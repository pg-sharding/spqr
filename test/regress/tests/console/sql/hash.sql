CREATE DISTRIBUTION ds1 COLUMN TYPES integer;

ALTER DISTRIBUTION ds1 ATTACH RELATION r1 DISTRIBUTION KEY col1 HASH FUNCTION IDENT;

ALTER DISTRIBUTION ds1 ATTACH RELATION r2 DISTRIBUTION KEY col1 HASH FUNCTION IDENTITY;

ALTER DISTRIBUTION ds1 ATTACH RELATION r3 DISTRIBUTION KEY col1 HASH FUNCTION MURMUR;

ALTER DISTRIBUTION ds1 ATTACH RELATION r4 DISTRIBUTION KEY col1 HASH FUNCTION CITY;

SHOW DISTRIBUTIONS;
SHOW RELATIONS;

DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;