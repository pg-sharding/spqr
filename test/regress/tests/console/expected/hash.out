
		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

ALTER DISTRIBUTION ds1 ATTACH RELATION r1 DISTRIBUTION KEY col1 HASH FUNCTION IDENT;
      attach table      
------------------------
 relation name   -> r1
 distribution id -> ds1
(2 rows)

ALTER DISTRIBUTION ds1 ATTACH RELATION r2 DISTRIBUTION KEY col1 HASH FUNCTION IDENTITY;
      attach table      
------------------------
 relation name   -> r2
 distribution id -> ds1
(2 rows)

ALTER DISTRIBUTION ds1 ATTACH RELATION r3 DISTRIBUTION KEY col1 HASH FUNCTION MURMUR;
      attach table      
------------------------
 relation name   -> r3
 distribution id -> ds1
(2 rows)

ALTER DISTRIBUTION ds1 ATTACH RELATION r4 DISTRIBUTION KEY col1 HASH FUNCTION CITY;
      attach table      
------------------------
 relation name   -> r4
 distribution id -> ds1
(2 rows)

SHOW DISTRIBUTIONS;
 Distribution ID | Column types 
-----------------+--------------
 ds1             | integer
(1 row)

SHOW RELATIONS;
 Relation name | Distribution ID |  Distribution key  
---------------+-----------------+--------------------
 r1            | ds1             | ("col1", identity)
 r2            | ds1             | ("col1", identity)
 r3            | ds1             | ("col1", murmur)
 r4            | ds1             | ("col1", city)
(4 rows)

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

